apps/api/src/app/import/import.service.ts

```diff
@@ -100,18 +101,67 @@ export class ImportService {
   }
 
   public async import({
+    accountsDto,
     activitiesDto,
     isDryRun = false,
     maxActivitiesToImport,
     userCurrency,
     userId
   }: {
+    accountsDto: Partial<CreateAccountDto>[];
     activitiesDto: Partial<CreateOrderDto>[];
     isDryRun?: boolean;
     maxActivitiesToImport: number;
     userCurrency: string;
     userId: string;
   }): Promise<Activity[]> {
+    const accountIdMapping = {};
+
+    //Create new accounts during dryRun so that new account IDs don't get invalidated
+    if (isDryRun && accountsDto?.length) {
+      for (let account of accountsDto) {
+        //Check if there is any existing account with the same ID
+        const accountWithSameId = await this.accountService.getAccountById(
```

Comment:
Instead of making database queries in the `for` loop, I would get all accounts from the user first with `const existingAccounts = await this.accountService.getAccounts(userId)` and in the loop you can use `existingAccounts.find(...)`

Reply:
Fixed as discussed